packages/ui/src/stack.tsx

```diff
@@ -12,6 +12,20 @@ export interface StackProps extends FlexProps {
   shouldWrapChildren?: boolean;
 }
 
+/**
+ * Gets only the valid children of a component,
+ * and ignores any nullish or falsy child.
+ *
+ * @param children the children
+ */
+export function getValidChildren(children: React.ReactNode) {
+  return React.Children.toArray(children).filter(child =>
+    React.isValidElement(child)
+  ) as React.ReactElement[];
+}
+
+export const selector = '& > *:not(:last-child)';
```

Comment:
You're not using this selector? 
https://github.com/chakra-ui/chakra-ui/blob/main/packages/layout/src/stack.utils.tsx#L12

Reply:
Yeah, I tried it and it works if we wanted to do spacing for basically `:not(:first-child)`, but that flips how things have been. Previously margins had been set to the right and bottom (for horizontal/vertical stack), and that selector would make it left, top. I noticed some interesting bugs where if certain things switched out, for a split second, spacing would be added to them. doing `:not(:last-child)` replicates the original behavior and works well in the wallet.

I'll test out `*:not(style):not(:last-child)` for SSR